### PR TITLE
Add --sync flag to copy-rds-snapshots to wait for copy actions to finish

### PR DESCRIPTION
**Pull request by mayitbeegh**
* Add --sync flag to copy-rds-snapshots to wait for copy actions to finish
* Add restore-rds-snapshot command

### CODE REVIEW

File: src/main/java/com/nike/cerberus/command/rds/RestoreRdsSnapshotCommand.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        commandNames = COMMAND_NAME,
+        commandDescription = "Restore RDS cluster snapshot in a specified region by updating cloudformation stack"
+)
+public class RestoreRdsSnapshotCommand implements Command {
+
+    public static final String COMMAND_NAME = "restore-rds-snapshot";
+
+    public static final String SNAPSHOT_IDENTIFIER = "--snapshot-identifier";
+
+    public static final String REGION_ENV_KEY = "RESTORE_RDS_SNAPSHOT_REGION";
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/command/rds/RestoreRdsSnapshotCommand.java]: This is confusing, can we only allow this to be passed in via command args and remove the logic for overloading it with env vars?

Comment (fieldju) [reply src/main/java/com/nike/cerberus/command/rds/RestoreRdsSnapshotCommand.java]: I don't think any other command follows this pattern

File: src/main/java/com/nike/cerberus/operation/composite/CompositeOperation.java [hunk 2]
```diff
@@ -115,6 +115,7 @@ public void run(T compositeCommand) {
         }
     }
 
+
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/operation/composite/CompositeOperation.java]: Remove this.

File: src/main/java/com/nike/cerberus/operation/rds/RestoreRdsSnapshotOperation.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    private final ConfigStore configStore;
+    private final String environmentName;
+    private final RdsService rdsService;
+    private final CloudFormationService cloudFormationService;
+    private final String DESIRED_STATE = "available";
+    private final String COPYING_STATE = "copying";
+
+    @Inject
+    public RestoreRdsSnapshotOperation(ConfigStore configStore,
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/operation/rds/RestoreRdsSnapshotOperation.java]: Why is this command / operation needed at all, when we already have https://github.com/Nike-Inc/cerberus-lifecycle-cli/blob/master/src/main/java/com/nike/cerberus/operation/core/UpdateStackOperation.java#L51 which supports updating the DB stack?

Comment (mayitbeegh) [reply src/main/java/com/nike/cerberus/operation/rds/RestoreRdsSnapshotOperation.java]: Nothing particular except for making it explicit and the environment variable override. I can take it out.

File: src/main/java/com/nike/cerberus/command/rds/RestoreRdsSnapshotCommand.java (hunk 1 above)
Comment (mayitbeegh) [reply src/main/java/com/nike/cerberus/command/rds/RestoreRdsSnapshotCommand.java]: It's the hacky way of potentially taking arguments from copy-snapshot command. Maybe there're better ways to do it or we can make the decision later.

Comment (fieldju) [reply src/main/java/com/nike/cerberus/command/rds/RestoreRdsSnapshotCommand.java]: OK, I remember talking about this with you. Maybe we should briefly look into extending the Composite operation to pass along state between sub commands.

Comment (mayitbeegh) [reply src/main/java/com/nike/cerberus/command/rds/RestoreRdsSnapshotCommand.java]: That's an option. We can make a decision when we chain them together.

Comment (mayitbeegh) [reply src/main/java/com/nike/cerberus/command/rds/RestoreRdsSnapshotCommand.java]: They're removed for now.

**Pull request merged**